File: 0069-sqrtx/0069-sqrtx.cpp
class Solution {
public:
    int mySqrt(int x) {
     if(x == 1) return 1;
     if(x == 0) return 0;
     long long  num=0;
    int ans=1;
     long long st = 1,end = x;
     while(st <= end){
      long long mid = st + (end-st)/2;
     long long num = (mid * mid);
      if(num == x) return mid;
      else if(num > x) end = mid-1;
      else{ 
        ans = mid;
        st = mid + 1;}
     }
     return ans;
}
};

File: 0069-sqrtx/0069-sqrtx.java
class Solution {
    public int mySqrt(int x) {
        //return (int)Math.sqrt(x);

        int st = 1, end = x,ans = 0;

        while(st<=end){
          int mid  = st + (end-st)/2;
          long num = (long)mid * mid;
          
           if(num <=x){
            ans = mid;
            st = mid+1;
          }
          else{
            end = mid - 1;
          }
        }
        return ans;
    }
}